src/links/links.service.ts

```diff
@@ -29,6 +32,54 @@ export class LinksService {
     }
   }
 
+  async menageShortLinkClick(
+    shortUrlCode: Link['shortUrlCode'],
+    ref = '',
+  ): Promise<string | null> {
+    try {
+      const referrer = ref.toLocaleLowerCase();
+      const link = await this.prisma.link.findUnique({
+        where: { shortUrlCode },
+        include: { linkStatistics: true },
+      });
+
+      const existingStatistic = link.linkStatistics.find(
+        (stat) => stat.referrer === referrer,
+      );
+
+      if (existingStatistic) {
+        await this.prisma.linkStatistic.update({
+          where: { id: existingStatistic.id },
+          data: {
+            clicks: { increment: 1 },
+          },
+        });
+      } else {
+        await this.prisma.linkStatistic.create({
+          data: {
+            referrer: ref,
+            clicks: 1,
+            link: {
+              connect: { id: link.id },
+            },
+          },
+        });
+      }
+
+      return link.originalUrl;
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw new NotFoundException(
+          `Link with shortUrlCode "${shortUrlCode}" not found.`,
+        );
+      } else {
+        throw new Error(
+          `Error fetching link with shortUrlCode "${shortUrlCode}"`,
+        );
+      }
+    }
+  }
+
   async findOrginalUrl(
     shortUrlCode: Link['shortUrlCode'],
   ): Promise<string | null> {
```

Comment:
**nitpick**: Consider removing or commenting out the logger statement if it's not needed in production.